src/Orleans/Core/ClusterClient.cs

```diff
@@ -0,0 +1,324 @@
+        }
+
+        /// <inheritdoc />
+        public IStreamProvider GetStreamProvider(string name)
+        {
+            this.ThrowIfDisposedOrNotInitialized();
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            return this.runtimeClient.CurrentStreamProviderManager.GetProvider(name) as IStreamProvider;
+        }
+
+        /// <inheritdoc />
+        public event ConnectionToClusterLostHandler ClusterConnectionLost
+        {
+            add
+            {
+                this.ThrowIfDisposed();
+                this.runtimeClient.ClusterConnectionLost += value;
+            }
+
+            remove
+            {
+                this.ThrowIfDisposed();
+                this.runtimeClient.ClusterConnectionLost -= value;
+            }
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="ClusterClient"/>, loading configuration using the default method.
+        /// </summary>
+        /// <returns></returns>
+        public static ClusterClient Create() => new ClusterClient(ClientConfiguration.StandardLoad());
+
+        /// <summary>
+        /// Creates a new <see cref="ClusterClient"/>, loading configuration from the specified file.
+        /// </summary>
+        /// <param name="configFilePath">The configuration file.</param>
+        /// <returns>A newly instantiated <see cref="ClusterClient"/>.</returns>
+        public static ClusterClient Create(string configFilePath) => Create(new FileInfo(configFilePath));
+
+        /// <summary>
+        /// Creates a new <see cref="ClusterClient"/>, loading configuration from the specified file.
+        /// </summary>
+        /// <param name="configFile">The configuration file.</param>
+        /// <returns>A newly instantiated <see cref="ClusterClient"/>.</returns>
+        public static ClusterClient Create(FileInfo configFile)
+        {
+            var config = ClientConfiguration.LoadFromFile(configFile.FullName);
+            if (config == null)
+            {
+                throw new ArgumentException(
+                    $"Error loading client configuration file {configFile.FullName}",
+                    nameof(configFile));
+            }
+
+            return new ClusterClient(config);
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="ClusterClient"/> using the provided configuration.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <returns>A newly instantiated <see cref="ClusterClient"/>.</returns>
+        public static ClusterClient Create(ClientConfiguration configuration) => new ClusterClient(configuration);
+
+        /// <inheritdoc />
+        public async Task Start()
+        {
+            this.ThrowIfDisposed();
+            using (await this.initLock.LockAsync())
+            {
+                await this.runtimeClient.Start();
```

Comment:
I added the disposal check after the lock (thanks). I'll add a double-start check.